components/style/stylist.rs

```diff
@@ -458,6 +471,38 @@ impl Stylist {
             .map(|(d, _)| d.invalidation_map.len()).sum()
     }
 
+    /// Set the registered property set associated with the document this
+    /// Stylist is styling.
+    pub fn set_registered_property_set(
+        &mut self,
+        registered_property_set: Arc<Locked<RegisteredPropertySet>>
+    ) {
+        debug_assert!(if let Some(ref old) = self.registered_property_set {
+            Arc::ptr_eq(old, &registered_property_set)
+        } else { true });
+        self.registered_property_set = Some(registered_property_set);
+    }
+
+    /// Get the registered property set associated with the document this
+    /// Stylist is styling. Panics if there is none set.
+    pub fn registered_property_set(&self) -> &Locked<RegisteredPropertySet> {
+        self.registered_property_set
+            .as_ref()
+            .map(|x| &**x)
+            .expect("set_registered_property_set should have been called.")
+    }
+
+    /// Returns whether the registered property set was updated since the last
+    /// time the Stylist was flushed.
+    pub fn registered_property_set_updated(&self, guard: &SharedRwLockReadGuard) -> bool {
+        let registered_property_set =
+            self.registered_property_set.as_ref()
+            .expect("set_registered_property_set must be called.")
+            .read_with(guard);
+
+        self.last_used_registered_property_set_generation != Some(registered_property_set.generation())
```

Comment:
There's nothing that sets `self.last_used_registered_property_set_generation`, right? That looks like a bug to me.

Reply:
Fell out while rebasing, it seems. Seems we can remove this per your comment below.